### ISSUE TITLE
mat-menu: button navigation on screen readers doesn't work while mat-menu is open

### ISSUE THREAD

**Issue by DJesseTaylor** [angular/components#23637]
#### Steps to reproduce:
This issue can be reproduced on the https://material.angular.io/components/menu/overview page with the first mat-menu example. 

1. Open a mat-menu item containing buttons while using a screen reader (Issue found while using JAWS)
![image](https://user-images.githubusercontent.com/8312896/134736783-f8d0c8db-85e1-46f9-814f-d3f8c61623d8.png)

2. Attempt to navigate to the next button by pressing b. (This is the command on JAWS it might be different on other screen readers)
 
#### Expected Behavior

Screen reader should focus the next button on the screen, which should be the first item in the action menu

#### Actual Behavior

Screen reader finds no buttons to navigate to.

#### Environment
- Angular/Material/CDK Version:
"@angular/cdk": "^12.2.3",
"@angular/common": "^12.2.3",
"@angular/compiler": "^12.2.3",
"@angular/core": "^12.2.3",
"@angular/forms": "^12.2.3",
"@angular/material": "^12.2.3",
- Browser(s): Chrome Version 93.0.4577.82
- Operating System (e.g. Windows, macOS, Ubuntu): Window
- Screen Reader: JAWS Professional Version 2020.1912.11

**Comment by zarend** [comment]
Hello [DJesseTaylor](https://github.com/DJesseTaylor),

This looks like another issue with Overlay popup being at the end of the DOM, rather than inline adjacent to the Overlay's trigger. What I believe is happening is that since focus is in the overlay popup, focus is at the end of the DOM. That means that trying to navigate to next button will give no result, since we have already passed all the buttons in the DOM by moving to the Overlay popup.

Best Regards,

Zach

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
